packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/text/TextLayoutManagerMapBuffer.java

```diff
@@ -487,6 +491,24 @@ public static long measureText(
             ? paragraphAttributes.getInt(PA_KEY_MAX_NUMBER_OF_LINES)
             : ReactConstants.UNSET;
 
+    Layout.Alignment alignment = Layout.Alignment.ALIGN_NORMAL;
+
+    MapBuffer fragments = attributedString.getMapBuffer(AS_KEY_FRAGMENTS);
+    if (fragments.getCount() != 0) {
+      MapBuffer fragment = fragments.getMapBuffer(0);
+      MapBuffer textAttributes = fragment.getMapBuffer(FR_KEY_TEXT_ATTRIBUTES);
+
+      if (textAttributes.contains(TextAttributeProps.TA_KEY_ALIGNMENT)) {
+        String alignmentAttr = textAttributes.getString(TextAttributeProps.TA_KEY_ALIGNMENT);
+
+        if (alignmentAttr.equals("center")) {
+          alignment = Layout.Alignment.ALIGN_CENTER;
+        } else if (alignmentAttr.equals("right")) {
+          alignment = Layout.Alignment.ALIGN_OPPOSITE;
+        }
+      }
+    }
```

Comment:
This is the same check as for `RTL`. When done this way it seems consistent with how it works on the old arch on Android (at least I didn't find differences), but differently than iOS.

The difference is in this case:
```jsx
    <Text style={{ backgroundColor: 'cyan' }}>
      <Text style={{ textAlign: props.textAlign }}>Parent Text</Text>
      <Text style={{ fontWeight: 'bold' }}>Child Text</Text>
      <View style={{width: 50, height: 50, backgroundColor: 'red'}} />
      <Text style={{ fontWeight: 'bold' }}>Child Text</Text>
      {props.long && <Text style={{ fontWeight: 'bold' }}>aaaa a aaaa aaaaaa aaa a a a aaaaa sdsds dsdSAD asd ASDasd ASDas</Text>}
    </Text>
```

Where the prop is set on the first inner text instead of the outer one. On iOS the set prop is applied to the entire paragraph, on Android it's not applied at all.

Reply:
Assuming this is ok, I will move it to a separate method and add implementation for `measureLines`. If it's not ok, what would be the best way to access the alignment from here?